src/main/java/com/glencoesoftware/omero/ms/image/region/ImageRegionMicroserviceVerticle.java

```diff
@@ -965,40 +975,102 @@ private void downloadFileAnnotation(RoutingContext event) {
+                        response.end("File for given Annotation does not exist");
+                        return;
+                    }
+                    if (file.isDirectory()) {
+                        response.setStatusCode(501);
+                        response.end("File Annotation of Unsupported File Type");
+                        return;
+                    }
+                    else {
+                        String contentType = "application/octet-stream";
+                        try {
+                            contentType = new Tika().detect(file);
+                        } catch (IOException e) {
+                            log.warn(String.format(
+                                    "Failed to detect content type of file %s",
+                                    fileName));
                         }
-                        if (file.isDirectory()) {
-                            response.setStatusCode(501);
-                            response.end("File Annotation of Unsupported File Type");
+                        response.headers().set("Content-Type", contentType);
+
+                        response.headers().set("Accept-Ranges", "bytes");
+                        ZonedDateTime zonedDateTime = ZonedDateTime.ofInstant(
+                                Instant.ofEpochMilli(file.lastModified()),
+                                ZoneId.of("UTC"));
+                        DateTimeFormatter formatter =
+                                DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+                        response.headers().set("Last-Modified", zonedDateTime.format(formatter));
+                        response.headers().set("Content-Disposition",
+                                "attachment; filename=\"" + fileName + "\"");
+
+                        if (request.method() == HttpMethod.HEAD) {
+                            response.headers().set("Content-Length", Long.toString(file.length()));
+                            response.end();
                             return;
                         }
-                        else {
-                            response.headers().set("Content-Type", "application/octet-stream");
-                            response.headers().set("Content-Disposition",
-                                    "attachment; filename=\"" + fileName + "\"");
+                        if (request.headers().contains("Range")) {
+                            String range = request.getHeader("Range");
+                            long start;
+                            long end;
+                            if (range.matches("^bytes=\\d+-\\d+$")) {
+                                String[] startEndStr =
+                                        range.substring("bytes=".length()).split("-");
+                                start = Long.valueOf(startEndStr[0]);
+                                end = Long.valueOf(startEndStr[1]);
+                                end = Math.min(end, file.length());
```

Comment:
I think this should be
```
end = Math.min(Long.valueOf(startEndStr[1]), file.length() - 1);
```

For a file with length and `Range: bytes=0-1000`, the body response is correct as 1000 is interpreted as the whole file length  but `Content-Range` is returned as `0-651/651`